templates/shift-scheduling/main.go

```diff
@@ -1,370 +1,248 @@
+
+	o.Version = schema.Version{
+		Sdk: sdk.VERSION,
+	}
 
-type shift struct {
-	Day    int       `json:"day"`
-	Worker workerID  `json:"worker"`
-	Type   shiftType `json:"type"`
-}
+	stats := statistics.NewStatistics()
 
-type constraint interface {
-	propagate(store.Store) []store.Change
-	isProvenInfeasible(s store.Store) bool
-}
+	result := statistics.Result{}
 
-type breakConstraint struct {
-	shifts        store.Domains
-	breakDuration int
-}
+	run := statistics.Run{}
 
-// newBreakConstraint create returns a new breakConstraint.
-func newBreakConstraint(shifts store.Domains, breakDuration int) constraint {
-	return &breakConstraint{
-		shifts:        shifts,
-		breakDuration: breakDuration,
-	}
-}
+	t := solution.RunTime().Seconds()
+	run.Duration = &t
+	result.Duration = &t
 
-// Propagate removes values from the neighboring domains if a domain
-// has been assigned to single value. That is, when a worker has been assigned
-// to a shift it can no longer be in the two shifts before and after.
-func (t *breakConstraint) propagate(s store.Store) []store.Change {
-	changes := make([]store.Change, 0)
+	nextShiftSolution := Output{}
 
-	for i := 0; i < t.shifts.Len(s); i++ {
-		domain := t.shifts.Domain(s, i)
+	nextShiftSolution.Status = "infeasible"
 
-		if v, singleton := domain.Value(); singleton {
-			for j := 1; j <= t.breakDuration; j++ {
-				if i-j > 0 {
-					changes = t.remove(s, i-j, v, changes)
-				}
-				if i+j < t.shifts.Len(s) {
-					changes = t.remove(s, i+j, v, changes)
-				}
-			}
+	if solution != nil && solution.HasValues() {
+		nextShiftSolution.Status = "suboptimal"
+		if solution.IsOptimal() {
+			nextShiftSolution.Status = "optimal"
 		}
-	}
 
-	return changes
-}
-
-// isProvenInfeasible returns true if the values in the domains can never
-// result in an acceptable solution assuming only more values can be removed
-// from the domains if the search would continue. That is, if we see two shifts
-// with the same worker assigned in any sequence of 3 shifts it is proven to
-// be infeasible. Also, if we see a shift for which there are no more values
-// in the domain it is proven infeasible.
-func (t *breakConstraint) isProvenInfeasible(s store.Store) bool {
-	for i := 0; i < t.shifts.Len(s); i++ {
-		domain := t.shifts.Domain(s, i)
-		// being empty can never result in a solution if we assume we can only
-		// remove more values (which there aren't any)
-		if domain.Empty() {
-			return true
-		}
-		if w1, singleton := domain.Value(); singleton {
-			for j := 1; j <= t.breakDuration; j++ {
-				if i-j > 0 && t.checkFeasible(s, w1, i-j) {
-					return true
-				}
-				if i+j < t.shifts.Len(s) && t.checkFeasible(s, w1, i+j) {
-					return true
+		nextShiftSolution.Value = solution.ObjectiveValue()
+		val := statistics.Float64(solution.ObjectiveValue())
+		result.Value = &val
+
+		usedWorkers := make(map[int]struct{})
+
+		for _, assignment := range assignments {
+			if solution.Value(x.Get(assignment)) >= 0.5 {
```

Comment:
`>= 0.9` is probably a bit close to what we want to do (protect against small numerical problems)

Reply:
changed this as well.